### DISCUSSION TITLE
DPRNN inference time

### DISCUSSION THREAD

**Discussion by manu0586** [asteroid-team/asteroid#455] [General]
Hi, 
I evaluate some models for speech enhancement using the python API (ConvTasNet,DPT,DPRNN and DCRN)
Did you already notice the big difference for time execution for the DPRNN model compared to other models ?

Used models: 

model_ConvTasNet = BaseModel.from_pretrained("JorisCos/ConvTasNet_Libri1Mix_enhsingle_16k") 
model_DPRNNTasNet = BaseModel.from_pretrained("JorisCos/DPRNNTasNet-ks2_Libri1Mix_enhsingle_16k")
model_DPTNet = BaseModel.from_pretrained("JorisCos/DPTNet_Libri1Mix_enhsingle_16k")
model_DCCRNet = BaseModel.from_pretrained("JorisCos/DCCRNet_Libri1Mix_enhsingle_16k")

inference : model.separate('test.wav')  test.wav is a  6 second recording @16kHz

here the time from a jupyter notebook for 'model.separate()' execution
ConvTasNet Wall time: 1.88 s
DPT              Wall time: 4.49 s
DCCRN         Wall time: 2.63 s
DPRNN        Wall time: 43.2 s

**Comment by mpariente** [comment]
Can you give some code for the "benchmark" so we see what you measured?

**Reply by manu0586** [reply]
I just compared the time execution for the model.separate() using the same 'test.wav' (6seconds @ 16kHz)

**Reply by mpariente** [reply]
On CPU or GPU?

**Reply by mpariente** [reply]
Which kernel size for DPRNN ad DTNet? 
The "just" doesn't work, information _is_ missing.

**Reply by manu0586** [reply]
I use a small GPU (Quadro P2000- 4Go)
I used the word  'Just' because all other information is hidden from user ^^
So for parameters information all is on the huggingface pages

https://huggingface.co/JorisCos/DPRNNTasNet-ks2_Libri1Mix_enhsingle_16k
https://huggingface.co/JorisCos/DPTNet_Libri1Mix_enhsingle_16k

But effectively the parameters for the filterbanks are very different : 
filterbank:
                      DPRNN   vs    DPT
  kernel_size:     2           vs   16
  n_filters:         64           vs  64
  stride:              1              vs  8

**Reply by mpariente** [reply]
That's the main source of larger execution time, the stride. 
You can expect at least x8 because of this.

**Comment by manu0586** [comment]
It seems that using model.separate() , the process is on CPU and not on GPU . Is it a way to force using GPU ?
I check that gpu is available with torch.cuda.is_available() and answer is true but monitoring with nvidia-smi show that the process is not on GPU

**Reply by mpariente** [reply]
Move the model to GPU?
